smack-experimental/src/test/java/org/jivesoftware/smackx/message_fastening/MessageFasteningElementsTest.java

```diff
@@ -0,0 +1,199 @@
+/**
+ *
+ * Copyright 2019 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.message_fastening;
+
+import static org.jivesoftware.smack.test.util.XmlUnitUtils.assertXmlSimilar;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.jivesoftware.smack.packet.StandardExtensionElement;
+import org.jivesoftware.smack.packet.Stanza;
+import org.jivesoftware.smack.packet.StanzaFactory;
+import org.jivesoftware.smack.packet.id.StandardStanzaIdSource;
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.test.util.TestUtils;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.message_fastening.element.ExternalElement;
+import org.jivesoftware.smackx.message_fastening.element.FasteningElement;
+import org.jivesoftware.smackx.message_fastening.provider.FasteningElementProvider;
+import org.jivesoftware.smackx.sid.element.OriginIdElement;
+
+import org.junit.jupiter.api.Test;
+
+public class MessageFasteningElementsTest {
+
+    private final StanzaFactory stanzaFactory = new StanzaFactory(new StandardStanzaIdSource());
+
+    /**
+     * Test XML serialization of the {@link FasteningElement} using the example provided by
+     * the XEP.
+     *
+     * @see <a href="https://xmpp.org/extensions/xep-0422.html#wrapped-payloads">XEP-0422 §3.1 Wrapped Payloads</a>
+     */
+    @Test
+    public void fasteningElementSerializationTest() {
+        String xml = "" +
```

Comment:
nit: superfluous `""` here.

Reply:
And in further places in this file.

Reply:
This is on purpose so that the XML is indented correctly and therefore readable :)

Reply:
I think if the newline would directly follow after `xml =`, the XML would be indented as well and the readability would even be increased (as there would be less noise).

Reply:
Not with my IDE. It would be auto indented to
```
        String xml =
                "<apply-to xmlns='urn:xmpp:fasten:0' id='origin-id-1' clear='true'>" +
                        "      <i-like-this xmlns='urn:example:like'>Very much</i-like-this>" +
                        "</apply-to>";
```

Reply:
I am sure your IDE allows to change the layout after it has applied auto indention. Or simply `sed s; "" +$;; MessageFasteningElementsTest.java`